controllers/openstackmachine_controller.go

```diff
@@ -311,6 +311,12 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 
 	openStackMachine.Status.InstanceState = &instance.State
 
+	address := []corev1.NodeAddress{{Type: corev1.NodeInternalIP, Address: instance.IP}}
+	if instance.FloatingIP != "" {
+		address = append(address, []corev1.NodeAddress{{Type: corev1.NodeExternalIP, Address: instance.FloatingIP}}...)
+	}
+	openStackMachine.Status.Addresses = address
```

Comment:
do you have a output comparing between before the change and after change?

Reply:
I will compare `spec.template.spec.floatingIP` with `openStackMachine.Status.Addresses[corev1.NodeExternalIP]` in the next PR.